### ISSUE TITLE
Training code

### ISSUE THREAD

**Issue by Tzy010822** [HKUNLP/Dream#2]
Great work!!! Do you have plans to release the training codes of Dream7B recently? Looking forward to your reply.

**Comment by jiacheng-ye** [comment]
Hi there, it's still in progress.

**Comment by dllm-reasoning** [comment]
Hey! We have recently released some SFT scripts for dLLMs as part of our own work. While these are for LLaDA, they can be modified to work with Dream as well: https://github.com/dllm-reasoning/d1

**Comment by razorback16** [comment]
any update on training/fine tuning script?

**Comment by SamuelMoor-Smith** [comment]
> any update on training/fine tuning script?

Also looking for an update :)

**Comment by allzero-kwon** [comment]
Any update for training code?

**Comment by lingjiechen2** [comment]
> Is there anyway that we could get a rough draft of the SFT code? I am trying to do some small fine-tuning based on the DiffuLLaMa code but seem to be breaking the model a little bit. Or an example of the loss function implementation?

@SamuelMoor-Smith


Hi there, we’ve built [dllm-trainer](https://github.com/ZHZisZZ/dllm-trainer), a lightweight finetuning framework for diffusion language models on top of the [🤗 Transformers](https://github.com/huggingface/transformers) `Trainer`. Give it a try if you’d like to finetune [LLaDA / LLaDA-MoE](https://arxiv.org/abs/2502.09992) and [Dream](https://arxiv.org/abs/2508.15487).
